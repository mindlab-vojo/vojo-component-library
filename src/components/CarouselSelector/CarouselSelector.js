import React, { useState } from 'react'
import PropTypes from 'prop-types'

import * as CarouselSelector from './CarouselSelector.style'

import ThemeConsumer from '../../style/ThemeManager/ThemeProvider'
import Typography from '../Typography'

const CarouselSelectorComponent = ({ items }) => {
  const [selectedItem, setSelectedItem] = useState(items[0])

  const handleSelectItem = (id) => {
    let currentItemIndex
    items.filter((item, index) => {
      if (item.id === id) {
        currentItemIndex = index
      }
    })
    setSelectedItem(items[currentItemIndex])
  }

  const renderCircles = (theme) => items.map(item => (
    <CarouselSelector.Circle
      key={item.id}
      onClick={() => handleSelectItem(item.id)}
      theme={theme}
      isSelected={item.id === selectedItem.id}
    >
      <CarouselSelector.TextWrapper>
        <Typography
          color={item.id === selectedItem.id ? theme.colors.whiteColor : theme.colors.blackColor}
          fontWeight="bolder">
          {item.title}
        </Typography>
      </CarouselSelector.TextWrapper>
    </CarouselSelector.Circle>
  ))

  const renderComponent = (theme) => (
    <CarouselSelector.Wrapper>
      <CarouselSelector.Carousel>
        {renderCircles(theme)}
      </CarouselSelector.Carousel>
      <CarouselSelector.Description>
        <Typography>
          {selectedItem.description}
        </Typography>
      </CarouselSelector.Description>
    </CarouselSelector.Wrapper>
  )

  return (
    <ThemeConsumer>
      {({ theme }) => renderComponent(theme)}
    </ThemeConsumer>
  )
}

CarouselSelectorComponent.propTypes = {
  items: PropTypes.array,
}

export default CarouselSelectorComponent
