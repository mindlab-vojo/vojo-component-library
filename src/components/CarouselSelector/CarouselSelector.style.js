import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  max-width: 100%;
`

export const Carousel = styled.div`
  overflow: auto;
  white-space: nowrap;
  width: 100%;

  &::-webkit-scrollbar {
    display: none;
  };
`

export const Circle = styled.div`
  align-items: center;
  background-color: ${props => props.isSelected ? props.theme.colors.primaryColor : '#e1e4e8' };
  border: 1px solid ${props => props.theme.colors.blackColor};
  border-radius: 100px;
  cursor: pointer;
  display: inline-block;
  justify-content: center;
  height: 47px;
  margin: 12px 5px;
  width: 47px;
`

export const TextWrapper = styled.div`
  align-items: center;
  display: flex;
  justify-content: center;
  height: 47px;
  width: 47px;
`

export const Description = styled.div``