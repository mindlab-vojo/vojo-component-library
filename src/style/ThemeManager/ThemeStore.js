import { vojo } from './../themes'

class ThemeStore {

  constructor(theme) {
    this.defaultTheme = vojo
    this.theme = theme ? theme : this.defaultTheme
  }

  configure = ({ theme }) =>  {
    if (theme) this.configureTheme(theme)
    else this.configureTheme(this.defaultTheme)
  }

  configureTheme = (theme) => {
    this.theme = theme
  }

  modify = (modifiedProps) => {
    const currentTheme = this.theme
    return { currentTheme, modifiedProps }
  }

  getTheme = () => this.theme

}

export default ThemeStore