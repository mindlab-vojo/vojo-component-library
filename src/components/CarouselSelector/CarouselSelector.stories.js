import React from 'react';
import { withKnobs, array, boolean, text } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';
import { action } from '@storybook/addon-actions'

import CarouselSelector from '.'; 
import { ThemeProvider } from "../../../src";
import categories from '../../../.storybook/categories';

export const carouselSelector = () => {  
  const itemsMock = [
    {
      "id": "MS",
      "title": "MS",
    },
    {
      "id": "SP",
      "title": "SP",
    },
    {
      "id": "RJ",
      "title": "RJ",
    },
    {
      "id": "SA",
      "title": "SA",
    },
    {
      "id": "WW",
      "title": "WW",
    },
    {
      "id": "ZZ",
      "title": "ZZ",
    },
    {
      "id": "QA",
      "title": "QA",
    },
    {
      "id": "MB",
      "title": "mb",
    },
  ]
  const description = text('description', "São Paulo, Rio de Janeiro, Rio Grande do Sul, Florianópolis, Belém, Uruguaiana")
  const items = array('items', itemsMock)
  const isLoading = boolean('isLoading', false)
  const onClickItem = action('onClickItem')

  return (
    <CarouselSelector
      description={description}
      isLoading={isLoading}
      items={items}
      onClickItem={onClickItem}
    />
  )
}

export default {
  title: categories.Dropdowns,
  name: 'CarouselSelector',
  component: carouselSelector,
  decorators: [
    withKnobs, 
    withA11y, 
    (story) => <ThemeProvider>{ story() }</ThemeProvider>,
    (story) => <div style={{ display: "flex" }}>{ story() }</div>,
  ],
}
  