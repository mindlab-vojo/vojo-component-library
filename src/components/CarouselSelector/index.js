import CarouselSelector from './CarouselSelector'
export default CarouselSelector