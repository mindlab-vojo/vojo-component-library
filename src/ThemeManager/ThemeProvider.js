import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  render() {
    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer