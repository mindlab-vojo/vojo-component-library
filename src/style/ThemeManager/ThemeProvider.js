import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  getThemeFont = () => {
    const { themeConfig } = this.state
    return themeConfig.font
  }

  render() {

    const GlobalFont = this.getThemeFont()

    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        <GlobalFont/>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer