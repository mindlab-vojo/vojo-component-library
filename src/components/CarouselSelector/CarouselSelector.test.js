import React from 'react';
import { render } from '@testing-library/react';

import CarouselSelector from './';
import { ThemeProvider } from './../../style/ThemeManager/ThemeProvider';

test('Test CarouselSelector component render', () => {
  const component = render(
    <ThemeProvider>
      <CarouselSelector
        items={[
          {
            "id": "MS",
            "title": "MS",
            "description": "Araquari, São José"
          },
          {
            "id": "SP",
            "title": "SP",
            "description": "São Paulo, Campinas, São José dos Campos, São Caetano do Sul"
          }
        ]}
      />
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
