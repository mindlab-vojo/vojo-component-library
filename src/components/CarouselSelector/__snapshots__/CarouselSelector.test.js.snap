// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CarouselSelector component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ jGJVBi"
      >
        <div
          class="sc-gsTCUz ddLIQr"
        >
          <div
            class="sc-dlfnbm dczYKs"
          >
            <div
              class="sc-hKgILt kceAIk"
            >
              <p
                class="sc-iBPRYJ kNbCVt"
              >
                MS
              </p>
            </div>
          </div>
          <div
            class="sc-dlfnbm gHGDmf"
          >
            <div
              class="sc-hKgILt kceAIk"
            >
              <p
                class="sc-fubCfw gjOGqP"
              >
                SP
              </p>
            </div>
          </div>
        </div>
        <div
          class="sc-eCssSg"
        >
          <p
            class="sc-pFZIQ WRJmx"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ jGJVBi"
    >
      <div
        class="sc-gsTCUz ddLIQr"
      >
        <div
          class="sc-dlfnbm dczYKs"
        >
          <div
            class="sc-hKgILt kceAIk"
          >
            <p
              class="sc-iBPRYJ kNbCVt"
            >
              MS
            </p>
          </div>
        </div>
        <div
          class="sc-dlfnbm gHGDmf"
        >
          <div
            class="sc-hKgILt kceAIk"
          >
            <p
              class="sc-fubCfw gjOGqP"
            >
              SP
            </p>
          </div>
        </div>
      </div>
      <div
        class="sc-eCssSg"
      >
        <p
          class="sc-pFZIQ WRJmx"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
